### PR TITLE
Simplify writeObject function

### PR DESCRIPTION
**Pull request by calixteman**
It'll avoid to have the duplication of the code to get the encrypt transform, and last but not least, it'll avoid to forget about encryption.

### CODE REVIEW

File: src/core/document.js [hunk 1]
```diff
@@ -321,16 +321,7 @@ class Page {
...
-    let transform = null;
-    if (this.xref.encrypt) {
-      transform = this.xref.encrypt.createCipherTransform(
-        this.ref.num,
-        this.ref.gen
-      );
-    }
-
-    await writeObject(this.ref, pageDict, buffer, transform);
+    await writeObject(this.ref, pageDict, buffer, this.ref);
```
Comment (Snuffleupagus) [inline src/core/document.js:324]: Given the typo in the last argument, if this passes tests as-is that'd unfortunately suggest that our test-coverage is slightly lacking here.
```suggestion
    await writeObject(this.ref, pageDict, buffer, this.xref);
```

**Review by Snuffleupagus — APPROVED** [review]
r=me, with the comment addressed and passing tests; thank you!

File: src/core/document.js (hunk 1 above)
Comment (calixteman) [reply src/core/document.js:324]: I just ran the unit tests locally.
That said this part should be covered by reftests.

Comment (calixteman) [reply src/core/document.js]: Here we just write the updated page dict which shouldn't contain any strings, streams, ... so the encryption is useless in this case.

**Pull request merged**